salt/modules/mysql.py

```diff
@@ -107,16 +113,22 @@
     "LOCK TABLES",
     "PERSIST_RO_VARIABLES_ADMIN",
     "PROCESS",
+    "READ_ONLY ADMIN",  # MariaDB since 10.5.2
     "REFERENCES",
     "RELOAD",
+    "REPLICA MONITOR",  # MariaDB since 10.5.9
     "REPLICATION CLIENT",
+    "REPLICATION MASTER ADMIN",  # MariaDB since 10.5.2
+    "REPLICATION REPLICA",  # MariaDB since 10.5.1
```

Comment:
Just found that it is not possible to apply this grant as it is an alias for `REPLICATION SLAVE` and the grant reported by MariaDB is `REPLICATION SLAVE` so when trying to apply `REPLICATION REPLICA` it fails because it is detected as a mismatch...

So to fully fix this it is required to add the information about grant aliases in this module.

Reply:
https://github.com/saltstack/salt/pull/59280 would be a good long term fix for this.

However i still think this pr would be good to merge now as it would at least restore functionality to actually use grants. It would be on the user to know which grants to use based on version until something like the above PR could be reworked